### PR TITLE
_setplotpencolor can now accept an array or value

### PR DESCRIPTION
**Pull request by pollend**
This is a pull request that resolves issue #540

### CODE REVIEW

File: src/main/org/nlogo/prim/plot/primitives.scala [hunk 1]
```diff
@@ -317,12 +317,25 @@ final class _setplotpenmode extends PlotCommand(Syntax.NumberType) {
...
 }
 
-final class _setplotpencolor extends PlotCommand(Syntax.NumberType) {
+final class _setplotpencolor extends PlotCommand(Syntax.NumberType | Syntax.ListType) {
   import org.nlogo.api.Color
   override def perform(context: Context) {
-    currentPen(context).color =
-      Color.getARGBbyPremodulatedColorNumber(
-        Color.modulateDouble(argEvalDoubleValue(context, 0)))
+    var obj = args(0).report(context);
```
Comment (SethTisue) [inline src/main/org/nlogo/prim/plot/primitives.scala]: no semicolons please, this is Scala

File: src/main/org/nlogo/prim/plot/primitives.scala [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 package org.nlogo.prim.plot
 
-import org.nlogo.api.{ CommandRunnable, I18N, Syntax }
+import org.nlogo.api.{ CommandRunnable, I18N, Syntax,LogoList }
```
Comment (SethTisue) [inline src/main/org/nlogo/prim/plot/primitives.scala]: inconsistent formatting

File: src/main/org/nlogo/prim/plot/primitives.scala (hunk 1 above)
Comment (SethTisue) [inline src/main/org/nlogo/prim/plot/primitives.scala]: also this should be a `val` not a `var`

Comment (SethTisue) [inline src/main/org/nlogo/prim/plot/primitives.scala]: come to think of it, you don't need the `val` either. just write:

```scala
args(0).report(context) match {
  ...
```

File: src/main/org/nlogo/prim/plot/primitives.scala [hunk 3]
```diff
@@ -317,12 +317,25 @@ final class _setplotpenmode extends PlotCommand(Syntax.NumberType) {
...
+        try
+          currentPen(context).color = Color.getARGBIntByRGBAList(rgbList)
+        catch {
+          case e: ClassCastException =>
+            throw new org.nlogo.nvm.EngineException(context, this, displayName + " an rgb list must contain only numbers")
+        }
+      case c: java.lang.Double => 
+        currentPen(context).color =
+          Color.getARGBbyPremodulatedColorNumber(
+          Color.modulateDouble(c))
```
Comment (SethTisue) [inline src/main/org/nlogo/prim/plot/primitives.scala]: incorrect indentation

**Comment by SethTisue** [comment]
is it possible to add test coverage for this? (genuine question, I don't recall what exists in this area)

also, surely the documentation needs updating as well?

**Comment by pollend** [comment]
https://ccl.northwestern.edu/netlogo/docs/dictionary.html#set-plot-pen-color

 If this is any indication, then yes the documentation needs to be updated.

File: src/main/org/nlogo/prim/plot/primitives.scala (hunk 3 above)
Comment (SethTisue) [inline src/main/org/nlogo/prim/plot/primitives.scala]: still not fixed in the new version

Comment (SethTisue) [inline src/main/org/nlogo/prim/plot/primitives.scala]: I'm suggesting replacing:

```scala
          Color.getARGBbyPremodulatedColorNumber(
          Color.modulateDouble(c))
```

with

```scala
          Color.getARGBbyPremodulatedColorNumber(
            Color.modulateDouble(c))
```

Comment (pollend) [inline src/main/org/nlogo/prim/plot/primitives.scala]: Ahh, that makes more sense. So if that one call had multiple lines would I keep it in line  Color.modulateDouble(c)

**Comment by mrerrormessage** [comment]
Thanks for the PR! At the moment the code on this branch doesn't compile and there's some trailing whitespace on 331. Additionally, the documentation will need to be updated `dist/docs/dictionary.html.mustache`. The `set-plot-pen-color` example should show "color" in italics instead of the current "number". Once these things have been addressed, I think this looks ready to go.

**Comment by mrerrormessage** [comment]
OK, I ran the tests for this and we have a failing test. In `sbt`, run `testOnly org.nlogo.compiler.TestAllSyntaxes` and fix the test, then this should be good to go.

**Comment by mrerrormessage** [comment]
Looks good, thanks!

**Pull request merged**